.github/workflows/example-builds-defaultarch.yml

```
@@ -44,3 +44,17 @@ jobs:
           version: ${{ matrix.julia-version }}
       - run: julia --version
       - run: julia --compile=min -O0 -e 'import InteractiveUtils; InteractiveUtils.versioninfo()'
+      - name: "Check that the correct julia is used and that archive mtimes are maintained"
+        run: |
+          if !occursin("hostedtoolcache", Sys.BINDIR)
+            error("the wrong julia is being used: $(Sys.BINDIR)")
+          end
+          if VERSION > v"1.7.0-0" # pkgdir was introduced here, and before then mtime wasn't a problem so just skip
```

Based upon the comment it sounds like this version includes `pkgdir`:
```suggestion
          if VERSION >= v"1.7.0-0" # pkgdir was introduced here, and before then mtime wasn't a problem so just skip
```

---

FWIW, you can make this a bit more concise: `if VERSION >= v"1.7.0-"`